Editor/src/Editor.bf

```diff
@@ -0,0 +1,615 @@
+			AddSetting(config, "DisplayWindowPadding", style.DisplayWindowPadding);
+			AddSetting(config, "DisplaySafeAreaPadding", style.DisplaySafeAreaPadding);
+			AddSetting(config, "MouseCursorScale", style.MouseCursorScale);
+			AddSetting(config, "AntiAliasedLines", style.AntiAliasedLines);
+			AddSetting(config, "AntiAliasedFill", style.AntiAliasedFill);
+			AddSetting(config, "CurveTessellationTol", style.CurveTessellationTol);
+			AddSetting(config, "CircleSegmentMaxError", style.CircleSegmentMaxError);
+			AddSettingType(config, "Colors", style.Colors);
+
+			var serialized = new String();
+
+			for (var prop in config)
+			{
+				if (prop.value.GetType() == typeof(ImGui.Vec2))
+				{
+					var vec = (ImGui.Vec2) prop.value;
+					serialized.AppendF("{} = [{}, {}]\n", prop.key, vec.x, vec.y);
+				}
+				else if (prop.value.GetType() == typeof(List<StringView>))
+				{
+					var list = (List<StringView>) prop.value;
+					serialized.AppendF("{} = [", prop.key);
+					for (var str in list)
+						serialized.AppendF("{}, ", str);
+					if (serialized.EndsWith(", "))
+						serialized.RemoveFromEnd(2);
+					serialized.[Friend]Realloc(serialized.AllocSize);
+					serialized.Append("]\n");
+				}
+				else
+				{
+					serialized.AppendF("{} = {}\n", prop.key, prop.value);
+				}
+			}
+
+			for (var value in config.Values)
+				delete value;
+			delete config;
+
+			var configPath = scope String();
+			SteelPath.GetEditorUserPath(configPath, "Config.txt");
+
+			if (File.WriteAllText(configPath, serialized) case .Err)
+				Log.Error("Failed to save style");
+
+			delete serialized;
+
+			void AddSetting<T>(Dictionary<StringView, Object> parent, StringView name, T value) where T : struct
+			{
+				parent[name] = new box value;
+			}
+
+			void AddSettingType(Dictionary<StringView, Object> parent, StringView name, ImGui.Vec4[(.) ImGui.Col.COUNT] vecArray)
+			{
+				var str = new String();
+				str.Append('[');
+				for (var vec in vecArray)
+					str.AppendF("[{}, {}, {}, {}], ", vec.x, vec.y, vec.z, vec.w);
+				str.RemoveFromEnd(2);
+				str.[Friend]Realloc(str.AllocSize);
+				str.Append("]");
+				parent[name] = str;
+			}
+		}
+
+		public static void LoadConfig()
+		{
+			var configPath = scope String();
+			SteelPath.GetEditorUserPath(configPath, "Config.txt");
+			var serialized = new String();
+			defer delete serialized;
+			if (File.ReadAllText(configPath, serialized) case .Err)
+				return;
+
+			ParseConfig(serialized, var config);
+			LoadStyle(config);
+
+			var windows = (List<Object>) config["Windows"];
+			for (var window in windows)
+				RegisterWindow((String) window);
+
+			DeleteConfig!(config);
+		}
+
+		public static void LoadStyle(StringView str)
+		{
+			ParseConfig(str, var config);
+			LoadStyle(config);
+			DeleteConfig!(config);
+		}
+
+		private static void LoadStyle(Dictionary<String, Object> config)
+		{
+			var style = ref ImGui.GetStyle();
+			style.Alpha = (float) config["Alpha"];
```

Comment:
Yeah, it was not intended to be handled like this, but because of limitations in the JSON library I couldn't make it work with structs, so I left it like this until structs were supported. Now that a new library has arrived (Which looks better) I might switch over to that.